internal/handlers/short_test.go

```diff
@@ -0,0 +1,137 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Parse short URLs", func() {
+	var (
+		imageID = "bf25292a-dddd-49dc-ab9c-3fb4c1f07071"
+
+		// generated at https://jwt.io/ with payload:
+		//
+		//	{
+		//		"infra_env_id": "bf25292a-dddd-49dc-ab9c-3fb4c1f07071"
+		//	}
+		tokenInfraEnv = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiJiZjI1MjkyYS1kZGRkLTQ5ZGMtYWI5Yy0zZmI0YzFmMDcwNzEifQ.VbI4JtIVcxy2S7n2tYFgtFPD9St15RrzQnpJuE0CuAI" //#nosec
+
+		// generated at https://jwt.io/ with payload:
+		//
+		//	{
+		//		"sub": "bf25292a-dddd-49dc-ab9c-3fb4c1f07071"
+		//	}
+		tokenSub = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpbmZyYV9lbnZfaWQiOiJiZjI1MjkyYS1kZGRkLTQ5ZGMtYWI5Yy0zZmI0YzFmMDcwNzEifQ.VbI4JtIVcxy2S7n2tYFgtFPD9St15RrzQnpJuE0CuAI" //#nosec
+
+		// generated at https://jwt.io/ with payload:
+		//
+		//	{
+		//	 	"name": "John Doe"
+		//	}
+		tokenNoID = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiSm9obiBEb2UifQ.DjwRE2jZhren2Wt37t5hlVru6Myq4AhpGLiiefF69u8" //#nosec
+	)
+
+	It("parse JWT with infra env ID", func() {
+		value, err := idFromJWT(tokenInfraEnv)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(value).To(Equal(imageID))
+	})
+
+	It("parse JWT with sub ID", func() {
+		value, err := idFromJWT(tokenSub)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(value).To(Equal(imageID))
+	})
+
+	It("parse invalid JWT", func() {
+		_, err := idFromJWT("not a JWT")
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("parse JWT that can't be base64 decoded", func() {
+		_, err := idFromJWT("foo.bar.baz")
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("parse JWT with no ID", func() {
+		_, err := idFromJWT(tokenNoID)
+
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("200 if infra_env_id present in token", func() {
```

Comment:
This test and the following ones should be in a `Describe("parseShortURL"...` block